qiskit/circuit/library/n_local/qaoa_ansatz.py

```diff
@@ -246,6 +246,10 @@ def mixer_operator(self, mixer_operator) -> None:
         self._mixer = mixer_operator
         self._invalidate()
 
+    @property
+    def num_qubits(self) -> int:
+        return self._cost_operator.num_qubits
+
```

Comment:
This will raise an error if `cost_operator` is not set yet, but perhaps we should return `None` instead?  I don't know what the behaviour of other similar classes is in these cases, though.

Reply:
Usually we return 0 in that case 🙂